ui/lib/pki/addon/components/pki-issuer-cross-sign.js

```diff
@@ -0,0 +1,173 @@
+ * intermediateIssuer (the name of the intermediate issuer, located in the above mount)
+ * newCrossSignedIssuer (the name of the to-be-cross-signed, new issuer)
+ *
+ * The requests involved and how those inputs are used:
+ * 1. Read an existing intermediate issuer
+ *    -> GET /:intermediateMount/issuer/:intermediateIssuer
+ * 2. Create a new CSR based on this existing issuer ID
+ *    -> POST /:intermediateMount/intermediate/generate/existing
+ * 3. Sign it with the new parent issuer, minting a new certificate.
+ *    -> POST /this.args.parentIssuer.backend/issuer/this.args.parentIssuer.issuerName/sign-intermediate
+ * 4. Import it back into the existing mount
+ *    -> POST /:intermediateMount/issuers/import/bundle
+ * 5. Read the imported issuer
+ *    -> GET /:intermediateMount/issuer/:issuer_id
+ * 5. Update this issuer with the newCrossSignedIssuer
+ *    -> POST /:intermediateMount/issuer/:issuer_id
+ *
+ * @example
+ * ```js
+ * <PkiIssuerCrossSign @parentIssuer={{this.model}} />
+ * ```
+ * @param {object} parentIssuer - the model of the issuing certificate that will sign the issuer to-be cross-signed
+ */
+
+export default class PkiIssuerCrossSign extends Component {
+  @service store;
+  @tracked formData = [];
+  @tracked signedIssuers = [];
+
+  inputFields = [
+    {
+      label: 'Mount path',
+      key: 'intermediateMount',
+      placeholder: 'Mount path',
+      helpText: 'The mount in which your new certificate can be found.',
+    },
+    {
+      label: "Issuer's current name",
+      key: 'intermediateIssuer',
+      placeholder: 'Current issuer name',
+      helpText: 'The API name of the previous intermediate which was cross-signed.',
+    },
+    {
+      label: 'New issuer name',
+      key: 'newCrossSignedIssuer',
+      placeholder: 'Enter a new issuer name',
+      helpText: `This is your new issuer’s name in the API.`,
+    },
+  ];
+
+  get statusCount() {
+    const error = this.signedIssuers.filter((issuer) => issuer.hasError).length;
+    const success = this.signedIssuers.length - error;
+    return `${success} successful, ${error} ${error === 1 ? 'error' : 'errors'}`;
+  }
+
+  @task
+  @waitFor
+  *submit(e) {
+    e.preventDefault();
+    this.signedIssuers = [];
+
+    // iterate through submitted data and cross-sign each certificate
+    for (let row = 0; row < this.formData.length; row++) {
+      const { intermediateMount, intermediateIssuer, newCrossSignedIssuer } = this.formData[row];
+      try {
+        // returns data from existing and newly cross-signed issuers
+        // { intermediateIssuer: existingIssuer, newCrossSignedIssuer: crossSignedIssuer, intermediateMount: intMount }
+        const data = yield this.crossSignIntermediate(
+          intermediateMount,
+          intermediateIssuer,
+          newCrossSignedIssuer
+        );
+        this.signedIssuers.addObject({ ...data, hasError: false });
+      } catch (error) {
+        this.signedIssuers.addObject({
```

Comment:
This could be a follow-on, but have we tested this workflow with various permissions? I wonder if this error format is going to make sense if they run into a permissions error on one of the earlier steps 🤔

Reply:
Definitely a great point to include in testing. I haven't tested but any errors that happen from making API requests in the `crossSignIntermediate` method are caught and should be added on line 92 with `hasError: errorMessage(error)`. Since permissions e

Reply:
🎉 
<img width="1292" alt="Screen Shot 2023-01-25 at 9 47 29 AM" src="https://user-images.githubusercontent.com/68122737/214642123-3dcff92a-eda7-46fd-ac77-ab5cf876e77c.png">

Reply:
Yay! One step better would be to know on which path the permission was denied, but at least we get something out of the box 🎉